### PR TITLE
Allow for linting in-memory dictionaries

### PR DESCRIPTION
**Pull request by jonhealy1**
https://github.com/stac-utils/stac-check/issues/91

### CODE REVIEW

**Review by philvarner — APPROVED** [review]

File: tests/test_lint.py [hunk 1]
```diff
@@ -287,3 +287,248 @@ def test_catalog_name():
...
+            "type": "application/geo+json",
+            "title": "Extended Item"
+            }
+        ]
+    }
+    linter = Linter(file)
+    assert linter.valid_stac == True
+    assert linter.asset_type == "COLLECTION"
+    assert linter.check_catalog_id_file_name() == False
+    assert linter.create_best_practices_dict()["check_catalog_id"] == ["Object should be called 'collection.json' not 'simple-collection.json'"]
```
Comment (gadomski) [inline tests/test_lint.py:402]: Collections usually have descriptive IDs, e.g. https://planetarycomputer.microsoft.com/api/stac/v1/collections/sentinel-2-l2a. I think this lint is incorrect.

**Review by gadomski — CHANGES_REQUESTED** [review]

File: tests/test_lint.py (hunk 1 above)
Comment (jonhealy1) [reply tests/test_lint.py:402]: This project was an attempt to follow the STAC Best Practices document. I think in the document this advice is only for static catalogs? https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#static-catalogs

Comment (jonhealy1) [reply tests/test_lint.py:402]: Here too: https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#catalog-layout

Comment (jonhealy1) [reply tests/test_lint.py:402]: Thanks for this. I'm changing the check so that it only activates with a static file.

**Review by gadomski — APPROVED** [review]

**Pull request merged**